src/profile-logic/comparison.js

```diff
@@ -858,3 +874,199 @@ function getComparisonThread(
+  } = combineResourceTables(translationMapsForLibs, newStringTable, threads);
+  const {
+    funcTable: newFuncTable,
+    translationMaps: translationMapsForFuncs,
+  } = combineFuncTables(translationMapsForResources, newStringTable, threads);
+  const {
+    frameTable: newFrameTable,
+    translationMaps: translationMapsForFrames,
+  } = combineFrameTables(
+    null,
+    translationMapsForFuncs,
+    newStringTable,
+    threads
+  );
+  const {
+    stackTable: newStackTable,
+    translationMaps: translationMapsForStacks,
+  } = combineStackTables(null, translationMapsForFrames, threads);
+
+  // Combine the samples for merging. This
+  const { samples: newSamples } = combineSamplesForMerging(
+    translationMapsForStacks,
+    threads
+  );
+
+  const { markerTable: newMarkers } = mergeMarkers(threads, newStringTable);
+
+  const processStartupTime = Math.min(
+    ...threads.map(thread => thread.processStartupTime)
+  );
+  const processShutdownTime = Math.max(
+    ...threads.map(thread => thread.processShutdownTime || Infinity)
+  );
+  const registerTime = Math.min(...threads.map(thread => thread.registerTime));
+  const unregisterTime = Math.max(
+    ...threads.map(thread => thread.unregisterTime || Infinity)
+  );
+
+  const mergedThread = {
+    processType: 'merged',
+    processStartupTime,
+    processShutdownTime: processShutdownTime === 0 ? null : processShutdownTime,
+    registerTime,
+    unregisterTime: unregisterTime === 0 ? null : unregisterTime,
+    pausedRanges: [],
+    name: 'Merged thread',
+    pid: 'Merged thread',
+    tid: undefined,
+    samples: newSamples,
+    markers: newMarkers,
+    stackTable: newStackTable,
+    frameTable: newFrameTable,
+    stringTable: newStringTable,
+    libs: newLibTable,
+    funcTable: newFuncTable,
+    resourceTable: newResourceTable,
+  };
+
+  return mergedThread;
+}
+
+/**
+ * This combines the sample tables for multiple threads.
+ * This is similar to combineSamplesDiffing function, but differently, this
+ * function adds all the samples as a positive value so they all add up in the end.
+ * And it does not handle different interval values since threads should belong to
+ * the same profile.
+ * It returns the new sample table with the translation maps to be used in
+ * subsequent merging functions, if necessary.
+ */
+function combineSamplesForMerging(
+  translationMapsForStacks: TranslationMapForStacks[],
+  threads: Thread[]
+): { samples: SamplesTable, translationMaps: TranslationMapForSamples[] } {
+  const translationMaps = [new Map(), new Map()];
+  const samples = threads.map(thread => thread.samples);
+  const sampleIndexes = Array(samples.length).fill(0);
+
+  const newSamples = getEmptySamplesTableWithEventDelay();
+
+  while (true) {
+    let selectedSamplesIndex: number | null = null;
+    let time = Infinity;
+    // 1. Find out which sample to consume.
+    for (let samplesIndex = 0; samplesIndex < samples.length; samplesIndex++) {
```

Comment:
It's not exactly O(n^2) because they iterate on different things. `while` iterates on all the samples combined so it can merge the samples. But inner `for` loop iterates over the sampleTables themselves.(like if we are merging 3 threads, this if will iterate 3 times). So it's like O(n*m) which in that case, n is total sample count, and m is thread count. It's better than n^2 but still probably there is a room for improvement. I added a todo that explains this.